### DISCUSSION TITLE
how to use this OSC extension with mpv_thumbnail_script?

### DISCUSSION THREAD

**Discussion by geextahslex** [christoph-heinrich/mpv-quality-menu#39] [Q&A]
Does someone know how to use this OSC extension with https://github.com/TheAMM/mpv_thumbnail_script?

Thank you :)

**Comment by christoph-heinrich** [comment]
You'll have to apply [this patch](https://github.com/christoph-heinrich/mpv-quality-menu/blob/master/osc.patch) to their `mpv_thumbnail_script_client_osc.lua`.

Any reason you're not using the updated https://github.com/marzzzello/mpv_thumbnail_script ?

**Reply by geextahslex** [reply]
>You'll have to apply this patch to their mpv_thumbnail_script_client_osc.lua.

Yes I have read that but I don't know how to do that. Thumbnail script has almost 4500 lines of code ^^'

>Any reason you're not using the updated https://github.com/marzzzello/mpv_thumbnail_script ?

The old one works fine at the moment

**Reply by christoph-heinrich** [reply]
> > You'll have to apply this patch to their mpv_thumbnail_script_client_osc.lua.
> 
> Yes I have read that but I don't know how to do that. Thumbnail script has almost 4500 lines of code ^^'

The patch has lines like `@@ -1615,6 +1615,13 @@ function bar_layout(direction)`.
The first number is the position in the original file, the second is the position after the changes that come before that change have already been applied. Keep in mind that those are only correct for the version of the osc that I made the patch from. That thumbnail script made a lot of changes, so those numbers are likely very wrong. But as you can see at the end of that line, that change is supposed to be in `function bar_layout(direction)`, so you find that function and then look for the correct position to add those lines in.
There is
```
     lo.geometry = geo
     lo.style = osc_styles.smallButtonsBar
```
above the change, and
```
     -- Volume
     geo = { x = geo.x - geo.w - padX, y = geo.y, an = geo.an, w = geo.w, h = geo.h }
     lo = add_layout("volume")
```
below it. So you look for the position that has both those parts and then simply paste the lines that have the `+` at the front (the `+` shouldn't be in the final result, so you have to remove that).

Simply repeat that with all sections in the patch.
